operator/main.go

```diff
@@ -509,14 +483,41 @@ func onOperatorStartLeading(ctx context.Context) {
 		}
 
 		if operatorOption.Config.SyncK8sNodes {
-			if err := runNodeWatcherKVStore(nodeManager); err != nil {
-				log.WithError(err).Error("Unable to setup node watcher")
-			}
+			cnWatcherCfgBuilder.WithKVStoreSync()
 		}
 
 		startKvstoreWatchdog()
 	}
 
+	cnWatcherCfg, err := cnWatcherCfgBuilder.Build()
+	if err != nil {
+		log.WithError(err).Fatal("Unable to generate configuration for CiliumNode watcher")
+	}
+	if err := startSynchronizingCiliumNodes(cnWatcherCfg); err != nil {
```

Comment:
I'm confused why the builder needs to exist here, why couldn't we just do something like:
```suggestion
	cnSync, err := NewCiliumNodeSynchronizer(cnWatcherCfg)
	if err != nil {
		log.WithError(err).Fatal("Unable to generate configuration for CiliumNode watcher")
	}
	if err := cnSync.startSynchronizingCiliumNodes(); err != nil {
```